packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleVideoViewer/SingleVideoViewerContainer.js

```diff
@@ -23,270 +23,182 @@ const DrawingLayer = styled.div`
   cursor: default;
 `
 
-class SingleVideoViewerContainer extends React.Component {
-  constructor() {
-    super()
-
-    this.videoViewer = React.createRef()
-    this.transformLayer = React.createRef()
-
-    this.state = {
-      vid: '',
-      isPlaying: false,
-      playbackRate: 1,
-      played: 0,
-      duration: 0,
-      isSeeking: false,
-      clientWidth: 0,
-      naturalWidth: 0,
-      naturalHeight: 0
+function SingleVideoViewerContainer({
+  enableInteractionLayer = true,
+  onError = () => true,
+  onReady = () => true,
+  onKeyDown = () => true,
+  subject
+}) {
+  const [clientWidth, setClientWidth] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isSeeking, setIsSeeking] = useState(false)
+  const [playbackRate, setPlaybackRate] = useState(1)
+  const [timeStamp, setTimeStamp] = useState(0)
+  const [videoHeight, setVideoHeight] = useState(0)
+  const [videoWidth, setVideoWidth] = useState(0)
+  const [videoSrc, setVideoSrc] = useState('')
+
+  const interactionLayerSVG = useRef()
+  const playerRef = useRef()
+  const transformLayer = useRef()
+
+  const { t } = useTranslation('components')
+
+  /* ==================== load subject ==================== */
+
+  useEffect(() => {
+    if (subject) {
+      setVideoSrc(Object.values(subject.locations[0])[0])
     }
-  }
-
-  componentDidMount() {
-    this.onLoad()
-  }
+  }, [subject])
 
-  /* ==================== get subject ==================== */
-  async onLoad() {
-    const { onError, onReady } = this.props
+  const onReactPlayerReady = () => {
     try {
-      const {
-        clientHeight,
-        clientWidth,
-        naturalHeight,
-        naturalWidth
-      } = await this.getVideoSize()
-      const target = { clientHeight, clientWidth, naturalWidth, naturalHeight }
-      this.setState({
-        naturalWidth: naturalWidth,
-        naturalHeight: naturalHeight
-      })
+      const reactPlayerVideoHeight = playerRef.current?.getInternalPlayer().videoHeight
+      const reactPlayerVideoWidth = playerRef.current?.getInternalPlayer().videoWidth
+      setVideoHeight(reactPlayerVideoHeight)
+      setVideoWidth(reactPlayerVideoWidth)
+
+      const { width: svgClientWidth, height: svgClientHeight } = interactionLayerSVG.current?.getBoundingClientRect()
```

Comment:
Destructuring here will crash with an error if the ref is null or undefined. Will that be a problem when drawing is disabled and there's no SVG element?

Reply:
This is also relates to your comment about if the svg layer should be rendered even when drawing tools aren't used. I was definitely was wondering the same thing. However, I think the svg layer is always rendered so that `onReady()` can be called with four dimensions. That function is a passed-down prop of SubjectViewerStore's `onSubjectReady()`, and the dimensions are saved in the store:

https://github.com/zooniverse/front-end-monorepo/blob/de3b6cce615f82d280caa805cd17b35973ace518/packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js#L96-L107

A similar pattern happens in SingleImageViewerContainer:

https://github.com/zooniverse/front-end-monorepo/blob/de3b6cce615f82d280caa805cd17b35973ace518/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js#L39-L48

Reply:
Subject images are always rendered as SVG, even if there's no drawing in the workflow, so that makes sense. For video, I'd have expected the client height and width to come from the video player. If the SVG does have to be rendered, and there's no drawing involved, I'd recommend styling it with `pointer-events: none;` so that it doesn't interfere with clicking on the video. I think that, at the moment, it's always positioned over the visible video and will capture the pointer.

Reply:
I guess the advantage of doing it this way is that the height and width of the player would include the controls, whereas the SVG seems to measure just the visible picture.

Reply:
Agreed with styling the pointer events. I think the height and width you mentioned actually exclude the controls because as of now, there are custom VideoController components rather than `react-player`'s built-in controls.

Do you know why the "client" and "natural" dimensions are saved in SubjectViewerStore? There's no mention of it in the store README's.

Reply:
It looks like those dimensions pertain to ClassificationMetadata as well:
https://github.com/zooniverse/front-end-monorepo/blob/de3b6cce615f82d280caa805cd17b35973ace518/packages/lib-classifier/src/store/Classification/ClassificationMetadata.js#L11-L16

Reply:
I think I can refactor `onReactPlayerReady()` to use client dimensions of its < video > element instead of relying on an SVG overlay being present. I'll work on that and let you know when ready for a new review!

Reply:
I refactored, and the drawing layers only render when `enableInteractionLayer` is true. It doesn't look like the passing of the `enableInteractionLayer` prop is built into the classifier yet, so I've set it as `false` while we're focused on Woodpecker Cavity Cam. `onReady()` is still called with the `target` expected in SubjectViewerStore.

Reply:
Client height and width are sent back with each classification. I don’t know if anyone has ever used them in their data analysis.

Reply:
https://github.com/zooniverse/front-end-monorepo/discussions/2118

Reply:
`enableInteractionLayer ` is built into the classifier. At least, it's used in transcription projects. I'm not 100% sure where it gets passed down. It might be one of the props that's passed in context.